internal/services/recommendation_poller.go

```diff
@@ -54,6 +54,7 @@ func transactionForRecommendation(recommendationSetList []model.RecommendationSe
 			return err
 		}
 	}
+	recommendationSuccess.Inc()
```

Comment:
As the `recommendationSuccess` metric says it is `The total number of recommendations saved by ROSOCP`, This increment should happen after committing the transaction.